tools/accuracy_checker/accuracy_checker/utils.py

```diff
@@ -518,3 +522,307 @@ def color_format(s, color=Color.PASSED):
+            'mxUINT8_CLASS': 'miUINT8',
+            'mxINT16_CLASS': 'miINT16',
+            'mxUINT16_CLASS': 'miUINT16',
+            'mxINT32_CLASS': 'miINT32',
+            'mxUINT32_CLASS': 'miUINT32',
+            'mxINT64_CLASS': 'miINT64',
+            'mxUINT64_CLASS': 'miUINT64'
+        }
+        self.inv_mclasses = dict((v, k) for k, v in self.mclasses.items())
+        self.compressed_numeric = ['miINT32', 'miUINT16', 'miINT16', 'miUINT8']
+
+    def read_file_header(self, fd, endian):
+        fields = [
+            ('description', 's', 116),
+            ('subsystem_offset', 's', 8),
+            ('version', 'H', 2),
+            ('endian_test', 's', 2)
+        ]
+        hdict = {}
+        for name, fmt, num_bytes in fields:
+            data = fd.read(num_bytes)
+            hdict[name] = self._unpack(endian, fmt, data)
+        hdict['description'] = hdict['description'].strip()
+        v_major = hdict['version'] >> 8
+        v_minor = hdict['version'] & 0xFF
+        hdict['__version__'] = '{}.{}'.format(v_major, v_minor)
+        return hdict
+
+    def read_var_header(self, fd, endian):
+        mtpn, num_bytes = self._unpack(endian, 'II', fd.read(8))
+        next_pos = fd.tell() + num_bytes
+
+        if mtpn == self.etypes['miCOMPRESSED']['n']:
+            data = fd.read(num_bytes)
+            dcor = zlib.decompressobj()
+            fd_var = BytesIO(dcor.decompress(data))
+            del data
+            fd = fd_var
+            if dcor.flush() != b'':
+                raise ParseError('Error in compressed data.')
+            mtpn, num_bytes = self._unpack(endian, 'II', fd.read(8))
+
+        if mtpn != self.etypes['miMATRIX']['n']:
+            raise ParseError('Expecting miMATRIX type number {}, '
+                             'got {}'.format(self.etypes['miMATRIX']['n'], mtpn))
+        header = self._read_header(fd, endian)
+        return header, next_pos, fd
+
+    def read_var_array(self, fd, endian, header):
+        mc = self.inv_mclasses[header['mclass']]
+
+        if mc in self.numeric_class_etypes:
+            return self._read_numeric_array(
+                fd, endian, header,
+                set(self.compressed_numeric).union([self.numeric_class_etypes[mc]])
+            )
+        if mc == 'mxSPARSE_CLASS':
+            raise ParseError('Sparse matrices not supported')
+        if mc == 'mxCHAR_CLASS':
+            return self._read_char_array(fd, endian, header)
+        if mc == 'mxCELL_CLASS':
+            return self._read_cell_array(fd, endian, header)
+        if mc == 'mxSTRUCT_CLASS':
+            return self._read_struct_array(fd, endian, header)
+        if mc == 'mxOBJECT_CLASS':
+            raise ParseError('Object classes not supported')
+        if mc == 'mxFUNCTION_CLASS':
+            raise ParseError('Function classes not supported')
+        if mc == 'mxOPAQUE_CLASS':
+            raise ParseError('Anonymous function classes not supported')
```

Comment:
I think that we need format if-statements (before all unsupported, then all supported)